### DISCUSSION TITLE
Add snapping & automerge to the Solvespace select tool

### DISCUSSION THREAD

**Discussion by Intuos5** [hlorus/CAD_Sketcher#142] [Feature Requests]
**Motivation / Existing Problem**
When moving point with the Solvespace select tool, I cannot modify the existing geometry to make it snap. 

https://user-images.githubusercontent.com/63149655/170325978-8db4c054-bd86-4fe2-b4ad-3a3307618be4.mp4

**Description**
When modifying sketches, one doesn't always apply constraints to make the sketch move. If I decided to join points in a hunch, it would be an intuitive thing to also be able to do this with the Solvespace select tool. 

**Alternative Solutions**
Use the coincident constraint applied to both points. 

**Mockup**
https://user-images.githubusercontent.com/63149655/170326726-746ba970-d834-40c4-b4ef-5efce7fc764f.mp4

**Comment by hlorus** [comment]
Would that only work with points or should it simply add a coincident constraint onto whatever is hovered?

**Reply by Intuos5** [reply]
Now that you mention it,  I checked in Fusion, it does coincident, midpoint and horizontal/ vertical constrains when moving points of a line.
Horizontal and vertical are applied even when the point is not snapped to other geometry. To me those all make sense,  so good thing you asked ;)
